sdk/identity/azure-identity/tests/test_cli_credential.py

```diff
@@ -239,8 +209,5 @@ def fake_check_output(command_line, **_):
         token = credential.get_token("scope")
         assert token.token == expected_token
 
-        with mock.patch.dict(
-            "os.environ", {EnvironmentVariables.AZURE_IDENTITY_DISABLE_MULTITENANTAUTH: "true"}
```

Comment:
I could be mistaken, but I don't think this test makes sense anymore without validating the behavior of setting this environment variable. Shouldn't we have a new test that checks for providing a tenant that's:
1. Not configured on the credential
2. Not in the list of allowed tenants

and verify that we raise a `ClientAuthenticationError` in that case?